instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/log/LogIOBenchmark.kt

```diff
@@ -0,0 +1,146 @@
+import androidx.benchmark.junit4.measureRepeated
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.datadog.android.Datadog
+import com.datadog.tools.unit.createInstance
+import com.datadog.tools.unit.forge.aThrowable
+import com.datadog.tools.unit.getFieldValue
+import com.datadog.tools.unit.invokeGenericMethod
+import com.datadog.tools.unit.invokeMethod
+import fr.xgouchet.elmyr.junit4.ForgeRule
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class LogIOBenchmark {
+
+    @get:Rule
+    val benchmark = BenchmarkRule()
+    @get:Rule
+    val forge = ForgeRule()
+
+    private lateinit var mockWebServer: MockWebServer
+
+    private lateinit var testedStrategy: Any
+    private lateinit var testedWriter: Any
+    private lateinit var testedReader: Any
+
+    @Before
+    fun setUp() {
+        mockWebServer = MockWebServer().apply { start() }
+        mockWebServer.setDispatcher(object : Dispatcher() {
+            override fun dispatch(request: RecordedRequest): MockResponse {
+                Thread.sleep(8000)
+                return mockResponse()
+            }
+        })
+        val fakeEndpoint = mockWebServer.url("/").toString().removeSuffix("/")
+
+        val context = InstrumentationRegistry.getInstrumentation().context
+        Datadog.initialize(context, "NO_TOKEN", fakeEndpoint)
+        testedStrategy = Datadog.getFieldValue("logStrategy")
+        testedWriter = testedStrategy.invokeMethod("getSynchronousWriter")!!
+        testedReader = testedStrategy.invokeMethod("getReader")!!
+    }
+
+    @After
+    fun tearDown() {
+        testedReader.invokeGenericMethod("dropAllBatches")
+        Datadog.invokeMethod("stop")
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun benchmark_write_logs_on_disk() {
+
+        benchmark.measureRepeated {
+            val log = runWithTimingDisabled { createLog() }
+            testedWriter.invokeGenericMethod("write", log)
+        }
+    }
+
+    @Test
+    fun benchmark_read_logs_from_disk() {
+        println("Writing logs")
+        for (i in 0..100) {
+            val log = createLog()
+            testedWriter.invokeGenericMethod("write", log)
+        }
+
+        println("Sleeping")
+        Thread.sleep(5500L)
+        println("Slept")
+
+        benchmark.measureRepeated {
+            println("Reading")
+            val batch = testedReader.invokeMethod("readNextBatch")
+            runWithTimingDisabled {
+                println("Read $batch")
+                checkNotNull(batch)
+                val id = batch.invokeMethod("getId") as String
+                testedReader.invokeMethod("releaseBatch", id)
```

Comment:
maybe you will also want to benchmark this part ? Maybe keep it outside of the closure ?